### ISSUE TITLE
Animated Gif Pause/Unpause

### ISSUE THREAD

**Issue by Wilson13** [facebook/fresco#1529]
Just wanted to be sure, is this function not supported or I overlooked it?

Thank you.

**Comment by massimocarli** [comment]
Hi @Wilson13 
at the moment we don't support the pause but we're working on it
Thanks

**Comment by Wilson13** [comment]
Hi @massimocarli

I see, looking forward to that function.

Thanks!

**Comment by liukangqi2015** [comment]
Hi @Wilson13 
On this subject, I have some ideas, this is my code
```
private Animatable mAnimatable;
DraweeController controller = Fresco.newDraweeControllerBuilder()
                .setUri(Uri.parse(url))
                .setAutoPlayAnimations(false)
                .setControllerListener(new BaseControllerListener<ImageInfo>(){
                    @Override
                    public void onFinalImageSet(String id, ImageInfo imageInfo, Animatable animatable) {
                        if (animatable!=null){
                            mAnimatable=animatable;
                        }
                    }
                })
                .build();
simpleDraweeView.setController(controller);
```
Then I use a button to control gif play or pause
```
btn.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                if (mAnimatable!=null){
                    if (isStart){
                        mAnimatable.start();
                        isStart=false;
                    }else {
                        mAnimatable.stop();
                        isStart=true;
                    }

                }
            }
        });
```

**Comment by oprisnik** [comment]
The issue with @liukangqi2015 's solution is that it doesn't pause but reset the animation. There is currently not a simple way to resume an ongoing animation without resetting. Our new animation architecture will support this though. We'll keep you updated when it's ready.

**Comment by liukangqi2015** [comment]
Looking forward to the update

**Comment by Wilson13** [comment]
Hi @liukangqi2015, 

@oprisnik was absolutely correct, and that was the issue I was facing. I couldn't pause then resume from where the animation has temporary stopped. It could only be stopped and restarted.

Thanks for your effort though =)

**Comment by liukangqi2015** [comment]
@Wilson13 ,sorry,on your question, I seem to understand the wrong.

**Comment by rafaelekol** [comment]
@oprisnik  looking forward for new Animation architecture

**Comment by razanbilwani** [comment]
@oprisnik is there a particular timeline for adding in the animation architecture? We're looking forward to it :smile:

**Comment by oprisnik** [comment]
Sorry, there's no particular timeline but we'll keep you updated.

**Comment by zetbrush** [comment]
Hey @oprisnik, any update on this?

**Comment by oprisnik** [comment]
The new architecture is out and I believe we recently made a change that should allow this to work. I'll check.

**Comment by jiansheliuxing** [comment]
Is there a better way to stop or restart the GIF?

**Comment by KimiChiu** [comment]
Hi, Any progress on it?

**Comment by oprisnik** [comment]
This has been fixed with 22f6426cf975a7f43ce9911895dede11e3eb0143, thanks to @yoavbeeri
